### PR TITLE
Support WalletConnect in gateway

### LINKED ISSUE #397: [gateway]: multi wallet support like wallet connect and others
## Issue Type

```
[ ] bug report
[x] feature request
```

## Expected Behavior

Support the Wallet Connect to let users use the multisig

## Steps to Reproduce

1.
2.
3.

### PR DESCRIPTION
**Pull request by boyuan-chen**
:clipboard: closed #397 

## Overview

Describe what your Pull Request is about in a few sentences.

Support WalletConnect in gateway

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- The new supported wallet is added to `wallet.service.js`
- Fix a crash issue on earn page on Boba Mainnet

image

## Testing

Describe how to test your new feature/bug fix and if possible, a step by step guide on how to demo this.

### CODE REVIEW

File: packages/boba/gateway/src/services/wallet.service.js [hunk 1]
```diff
@@ -15,56 +15,150 @@ See the License for the specific language governing permissions and
...
+      // so we need to add the code of adding chain.
+      await provider.request({
         method: "wallet_switchEthereumChain",
         params: [{ chainId }],
       })
     } catch (error) {
+      await provider.request({
+        method: "wallet_addEthereumChain",
+        params: [chainInfo, this.account],
+      })
```
Comment (sk-enya) [inline packages/boba/gateway/src/services/wallet.service.js]: This method will get triggered 2 times for meta mask.

**Review by sk-enya — APPROVED** [review]

File: packages/boba/gateway/src/services/wallet.service.js (hunk 1 above)
Comment (boyuan-chen) [reply packages/boba/gateway/src/services/wallet.service.js]: I can't reproduce the error. How can I reproduce it?

**Comment by sk-enya** [comment]
On setting up with fresh meta mask failing with error on switch to avax on wrong network modal.

Screenshot 2023-03-08 at 5 18 15 PM

**Comment by sk-enya** [comment]
@Boyuan-Chen can you please help me ? how we can test out the wallet connect on test network!

**Comment by boyuan-chen** [comment]
> On setting up with fresh meta mask failing with error on switch to avax on wrong network modal.
> 
> Screenshot 2023-03-08 at 5 18 15 PM

Could you please try again? The block explorer url is fixed.

**Comment by boyuan-chen** [comment]
> @Boyuan-Chen can you please help me ? how we can test out the wallet connect on test network!

I use onekey mac desktop app to test it. The gnosis safe doesn't support the feature of switching the network.

**Comment by sk-enya** [comment]
> > @Boyuan-Chen can you please help me ? how we can test out the wallet connect on test network!
> 
> I use onekey mac desktop app to test it. The gnosis safe doesn't support the feature of switching the network.

Okay, so any idea like how do we educate user correctly as we don't have control on switching network there?

**Comment by boyuan-chen** [comment]
> > > @Boyuan-Chen can you please help me ? how we can test out the wallet connect on test network!
> > 
> > 
> > I use onekey mac desktop app to test it. The gnosis safe doesn't support the feature of switching the network.
> 
> Okay, so any idea like how do we educate user correctly as we don't have control on switching network there?

If they pick wallet connect, they should know what they are doing. It's more complicated than MetaMask. 
In onekey wallet, the walletconnect supports the network switch and everything works fine. The Gnosis is different, because it only supports Ethereum L1 and there is no Gnosis on Boba L2 (Offical Gnosis) so that switching network from L1 to L2 is impossible.

**Review by wsdt — APPROVED** [review]

**Pull request merged**